### PR TITLE
Improve executor messages

### PR DESCRIPTION
**Pull request by michal367**
changes:
- real time remote logs
- ExecutorLoggers
- information about compressing, extracting and transferring in remote
- information about executing TRiP98 (remote and local)
- fix bug that was causing crash of execute sometimes (reason: using rstrip instead of splitext)
- remove os.chdir calls

associated with pytrip/pytripgui#365
associated with pytrip/pytripgui#366
associated with pytrip/pytripgui#510

### CODE REVIEW

**Comment by grzanka** [comment]
@michal367 can you post a screenshot here how improvement logger looks like ?

also - can Deepsource issues be fixed ?

**Comment by michal367** [comment]
> 
> 
> @michal367 can you post a screenshot here how improvement logger looks like ?
> 
> also - can Deepsource issues be fixed ?

I've added screenshot on pytripgui here: https://github.com/pytrip/pytripgui/pull/537

File: pytrip/tripexecuter/execute.py [hunk 1]
```diff
@@ -153,7 +169,16 @@ def _pre_execute(self, plan):
         if not plan.working_dir:
             logger.warning("No working directory was specified for plan. Setting it to ./")
             plan.working_dir = "./"
-        plan._working_dir = os.path.expandvars(plan.working_dir)
+        self._working_dir = os.path.expandvars(plan.working_dir)
+
+        # create default loggers that print output to console and files
+        if self._use_default_loggers is True:
```
Comment (rozchlastywacz) [inline pytrip/tripexecuter/execute.py:175]: more pythonic:
`if self._use_default_loggers:
(...)`

File: pytrip/tripexecuter/execute.py [hunk 2]
```diff
@@ -597,3 +657,35 @@ def test_remote_trip(self):
...
+        """
+        if self._use_default_loggers is True:
+            self.file_logger.end()
+        for executor_logger in self.executor_loggers:
+            executor_logger.end()
+
+    def _loggers(self, func_name, text):
+        """ Sends text to all executor loggers
+        """
+        if self._use_default_loggers is True:
```
Comment (rozchlastywacz) [inline pytrip/tripexecuter/execute.py:686]: more pythonic:
`if self._use_default_loggers:
(...)`

**Review by rozchlastywacz — COMMENTED** [review]
More comments about whats happening in the code :)
0983e2a55e71984829ef45a4e3f9f315

File: pytrip/tripexecuter/execute.py [hunk 3]
```diff
@@ -303,25 +295,19 @@ def _run_trip_remote(self, plan, _run_dir=None):
...
 
         tar_path = self._compress_files(plan._temp_dir)  # make a tarball out of the TRiP98 package
 
         # prepare relevant dirs and paths
         _, tgz_filename = os.path.split(tar_path)
+        local_tgz_path = os.path.join(self._working_dir, tgz_filename)
         remote_tgz_path = os.path.join(self.remote_base_dir, tgz_filename)
-        local_tgz_path = os.path.join(plan._working_dir, tgz_filename)
-        remote_run_dir = os.path.join(self.remote_base_dir, tgz_filename.rstrip(".tar.gz"))  # TODO: os.path.splitext
+        remote_rel_run_dir = os.path.splitext(os.path.splitext(tgz_filename)[0])[0]  # remove .tar.gz
```
Comment (rozchlastywacz) [inline pytrip/tripexecuter/execute.py:308]: What about using a variable instead of `very_long_method()[0]`?

**Review by rozchlastywacz — APPROVED** [review]
Looks nice 
[](https://c.tenor.com/hnP8FWI5fB4AAAAC/joseph-joestar-nice.gif)

**Review by grzanka — APPROVED** [review]
Looks good

**Pull request merged**